src/PostRector/Application/PostFileProcessor.php

```diff
@@ -126,6 +128,7 @@ private function getPostRectors(): array
             $postRectors[] = $this->unusedImportRemovingPostRector;
         }
 
+        $postRectors[] = $this->enumReferenceUpdaterPostRector;
```

Comment:
Add new `PostRector` probably need check if `RenamedEnumCaseCollector->hasNames()` so no need to traverse `EnumReferenceUpdaterPostRector` when the rule not executed.

Not sure if `PostRector` actually needed tho, define multiple `node` possibly enough:

```diff
    public function getNodeTypes(): array
    {
-        return [Enum_::class];
+        return [Enum_::class, ClassConstFetch::class];
    }
```

Then, in refactor, verify:

```diff
    /**
-    * @param Enum_ $node
+    * @param Enum_|ClassConstFetch $node
     */
    public function refactor(Node $node): ?Node
    {
+          if ($node instanceof ClassConstFetch) {
+               // process with RenamedEnumCaseCollector here...
+          }

           // process with Enum_ here ...
    }
```

Reply:
PostRectors are limited only for internal use only. Should not be touched, unless fixing something else.

Reply:
@TomasVotruba that's what I suggest on code diff suggestion above with define multiple nodes ( `Enum_::class, ClassConstFetch::class` ) in the rector rule itself.

Reply:
I used that kind of rule also as first idea but would have the problem, that i would change all Enum cases regardingless if its in a 3rd party vendor directory or not? Or i want to use that rule only for a specific module (like the e2e Test) i should not rename all enum cases i will find? 

The goal should be, that only that Enum cases are modified in the code that are really touched.

Reply:
On summary, when changing `ClassConstFetch`, verify that the target Enum file ( get via ReflectionProvider) is on target "filePaths"  (the files that rector try to change) so safer,

Reply:
ok, if Target Enum file is 3rd party, then it should only verify renamed ClassConstFetch exists via ClassReflection that you can pull from ReflectionProvider